finality-aleph/src/environment.rs

```diff
@@ -430,80 +419,37 @@ where
             )),
         );
     }
-}
 
-impl<B, H, C, BE, SC> Unpin for Environment<B, H, C, BE, SC>
-where
-    B: Block,
-    H: Hash,
-    C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
-    BE: Backend<B> + 'static,
-    SC: SelectChain<B> + 'static,
-{
-}
-
-impl<B: Block, H: Hash, C, BE, SC> Future for Environment<B, H, C, BE, SC>
-where
-    B: Block,
-    H: Hash,
-    C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
-    BE: Backend<B> + 'static,
-    SC: SelectChain<B> + 'static,
-{
-    type Output = ();
-    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-        debug!(target: "env", "Polling environment.");
-        if self.rx_consensus.is_some() {
-            loop {
-                match self.rx_consensus.as_mut().unwrap().poll_next_unpin(cx) {
-                    Poll::Ready(Some(notification)) => {
-                        self.on_consensus_notification(notification);
-                    }
-                    Poll::Ready(None) => {
+    pub(crate) async fn run_epoch(mut self) {
+        let mut request_ticker = time::interval(TICK_INTERVAL).fuse();
+        loop {
+            futures::select! {
```

Comment:
You can take a look at tokio::select! -- it does not require fused futures. Not sure it matters though.

Reply:
Good to know! I will keep it that way, one fuse for a local variable seems acceptable